lib/core/src/Cardano/Wallet/Primitive/Types.hs

```diff
@@ -694,6 +696,28 @@ instance FromText PoolId where
             , "bytes in length."
             ]
 
+-- | Encode 'PoolId' as Bech32 with "pool" hrp.
+encodePoolIdBech32 :: PoolId -> T.Text
```

Comment:
**Request:** could you add a roundtrip encoding/decoding test for `encodePoolIdBech32` and `decodePoolIdBech32` (see https://github.com/input-output-hk/cardano-wallet/pull/2055#pullrequestreview-473111599).

I'm guessing that the following property should hold:
```
decodePoolIdBech32 (encodePoolIdBech32 p) == Right p
```

We do have similar tests from which you can draw inspiration: have a search for `textRoundtrip`.

Feel free to ask if I can be of any assistance!